source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/EndpointTests/OutboxMessageProcessorEndpointTests.cs

```diff
@@ -107,53 +141,100 @@ public async Task RunAsync_WhenOutboxMessageIsRead_AvailableDataIsPersisted_AndP
                 await using var chargesDatabaseWriteContext = _databaseManager.CreateDbContext();
                 await using var chargesDatabaseReadContext = _databaseManager.CreateDbContext();
 
+                var rejectedEvent = rejectedEventBuilder.Build();
                 clock.Setup(c => c.GetCurrentInstant()).Returns(now);
+                outboxMessageParser
+                    .Setup(o => o.Parse(It.IsAny<string>(), It.IsAny<string>()))
+                    .Returns(rejectedEvent);
                 var operationsRejectedEvent = CreateChargePriceOperationsRejectedEvent();
```

Comment:
Not sure why there is two PriceRejectionEvents in play here, couldn't we just do this?
```suggestion
                clock.Setup(c => c.GetCurrentInstant()).Returns(now);
                var operationsRejectedEvent = CreateChargePriceOperationsRejectedEvent();
                outboxMessageParser
                    .Setup(o => o.Parse(It.IsAny<string>(), It.IsAny<string>()))
                    .Returns(operationsRejectedEvent);
```

Reply:
Done